use driver internal function in falcon_gpio_irq_type (instead gpio_direction_input)
to avoid dump in gpio_ensure_requested()

--- a/drivers/gpio/gpio-falcon.c
+++ b/drivers/gpio/gpio-falcon.c
@@ -187,7 +187,7 @@ static int falcon_gpio_irq_type(struct i
 		port_w32_mask(itop(d), mask, 0, GPIO_EXINTCR1);
 	}
 
-	return gpio_direction_input(itop(d)->gpio_chip.base + offset);
+	return falcon_gpio_direction_input(&itop(d)->gpio_chip, offset);
 }
 
 static void falcon_gpio_irq_handler(struct irq_desc *desc)
